strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/GroupRepositoryProvider.java

```diff
@@ -59,43 +52,60 @@
 
     @Inject
     private GroupRepositorySetCollector groupRepositorySetCollector;
-    
+
     @PersistenceContext
     private EntityManager entityManager;
-    
+
     @Inject
     private RepositoryPathResolver repositoryPathResolver;
-    
+
+    private ExecutorService executorService;
+
+    @Override
+    public void afterPropertiesSet()
+    {
+        executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
```

Comment:
this will will cause limited group repositories concurrency and request blocking, as you noticed. 
if you don't want to use `Collection.parallelStream()`, like was suggested, then can you please use `EventsConfig.eventTaskExecutor`reference here instead?

Reply:
to be clear I mean that `Collection.parallelStream()` will not completely block other requests because it use caller thread, so at least one thread will be available.
other option is  `EventsConfig.eventTaskExecutor` which use Servlet Container thread pool, so concurrency will be totally limited with our application performance.
as for me then I would like to try `Collection.parallelStream()` and see how it will work in production.

Reply:
> to be clear I mean that Collection.parallelStream() will not completely block other requests because it use caller thread, so at least one thread will be available.

I'm worried about blocking all threads from servlet container pool.

> EventsConfig.eventTaskExecutor which use Servlet Container thread pool, 

The same here

Reply:
As you said, `Collection.parallelStream() ` will utilize caller thread too (a thread from servlet container pool). If 10 independent requests arrive to the server then 10 threads from the servlet container pool could be utilized and blocked here.